### DISCUSSION TITLE
Changing default seralization of extensions in the StarterKit

### DISCUSSION THREAD

**Discussion by yannickgloster** [aguingand/tiptap-markdown#29] [Q&A]
Discord doesn't follow a proper markdown spec unfortunately but I'm looking to be able to generate text that is compatible with Discord.

I need to change the behavior of HardBreak and new lines as Discord treats multiple line quotes not needing a blank >. As well, HardBreak introduces `\` which isn't supported by Discord.

I'm unable to figure out how to extend the specific packages in StaterKit. Would you know how to go about doing that?

I've been able to update Underline as successfully as such:
```js
Underline.extend({
  addStorage() {
    return {
      markdown: {
        serialize: {
          open: "__",
          close: "__",
        },
      },
    };
  },
}),
```

**Comment by ctjlewis** [comment]
@aguingand I came here to ask about something similar - I'm using the default Mentions extension but the Mention nodes are serialized to Markdown as the static string `"[mention]"`. `editor.state.toText()` returns it correctly as `@name`.

Any advice on overriding how nodes are serialized? Thanks OP for this post also.

<img width="573" alt="image" src="https://github.com/aguingand/tiptap-markdown/assets/1657236/812b9f51-0911-496f-930b-35b9858757f0">

**Comment by aguingand** [comment]
You can override default behavior by using `.extend()` and defining a markdown spec (parse + serialize) for needed nodes / marks as described [here](https://github.com/aguingand/tiptap-markdown#custom-extensions).

While this package allows customization, its philosophy is to render standardized markdown.
